### ISSUE TITLE
vpp-smooth + fp16 -> green image

### ISSUE THREAD

**Issue by Selur** [rigaya/NVEnc#295]
Using:
```
NVEncC (x64) 5.24 (r1764) by rigaya, Dec 30 2020 12:27:24 (VC 1928/Win/avx2)
  [NVENC API v11.0, CUDA 10.1]
 reader: raw, y4m, avi, avs, vpy, avsw, avhw [H.264/AVC, H.265/HEVC, MPEG2, VP8, VP9, VC-1, MPEG1, MPEG4, AV1]

Environment Info
OS : Windows 10 x64 (19042)
CPU: AMD Ryzen 7 1800X Eight-Core Processor (8C/16T)
RAM: Used 6382 MB, Total 32697 MB

#0: GeForce GTX 1070 Ti (2432 cores, 1683 MHz)[PCIe3x16][461.09]
NVEnc features
Codec: H.264/AVC
Encoder Engines           2
Max Bframes               4
B Ref Mode                2 (only middle)
RC Modes                  63
Field Encoding            1 (field mode)
MonoChrome                no
FMO                       no
Quater-Pel MV             yes
B Direct Mode             yes
CABAC                     yes
Adaptive Transform        yes
Max Temporal Layers       4
Hierarchial P Frames      no
Hierarchial B Frames      no
Max Level                 62 (6.2)
Min Level                 10 (1)
4:4:4                     yes
Min Width                 33
Max Width                 4096
Min Height                17
Max Height                4096
Multiple Refs             no
Max LTR Frames            8
Dynamic Resolution Change yes
Dynamic Bitrate Change    yes
Forced constant QP        yes
Dynamic RC Mode Change    no
Subframe Readback         yes
Constrained Encoding      yes
Intra Refresh             yes
Custom VBV Bufsize        yes
Dynamic Slice Mode        yes
Ref Pic Invalidiation     yes
PreProcess                no
Async Encoding            yes
Max MBs                   65536
Lossless                  yes
SAO                       no
Me Only Mode              1 (I,P frames)
Lookahead                 yes
AQ (temporal)             yes
Weighted Prediction       yes
10bit depth               no

Codec: H.265/HEVC
Encoder Engines           2
Max Bframes               0
B Ref Mode                0 (no)
RC Modes                  63
Field Encoding            0 (no)
MonoChrome                no
Quater-Pel MV             yes
B Direct Mode             no
Max Temporal Layers       0
Hierarchial P Frames      no
Hierarchial B Frames      no
Max Level                 186 (6.2)
Min Level                 30 (1)
4:4:4                     yes
Min Width                 65
Max Width                 8192
Min Height                33
Max Height                8192
Multiple Refs             no
Max LTR Frames            7
Dynamic Resolution Change yes
Dynamic Bitrate Change    yes
Forced constant QP        yes
Dynamic RC Mode Change    no
Subframe Readback         yes
Constrained Encoding      yes
Intra Refresh             yes
Custom VBV Bufsize        yes
Dynamic Slice Mode        yes
Ref Pic Invalidiation     yes
PreProcess                no
Async Encoding            yes
Max MBs                   262144
Lossless                  yes
SAO                       yes
Me Only Mode              1 (I,P frames)
Lookahead                 yes
AQ (temporal)             no
Weighted Prediction       yes
10bit depth               yes


NVDec features
  H.264/AVC:  nv12, yv12
  H.265/HEVC: nv12, yv12, yv12(9bit), yv12(10bit), yv12(12bit)
  MPEG1:      nv12, yv12
  MPEG2:      nv12, yv12
  MPEG4:      nv12, yv12
  VP8:        nv12, yv12
  VP9:        nv12, yv12
  VC-1:       nv12, yv12
```
with:
```
NVEncC64 --avhw  -i "F:\TestClips&Co\files\5000frames.mp4" --fps 25.000 --codec h264 --profile high --level auto --sar 1:1 --lookahead 16 --vbrhq 0 --vbr-quality 0.00 --max-bitrate 17500 --gop-len 0 --ref 3 --bframes 3 --direct auto --bref-mode disabled --no-b-adapt --mv-precision Q-pel --cabac --deblock --preset default --colormatrix bt470bg --vpp-resize default --output-res 640x480 --vpp-gauss disabled --vpp-smooth quality=3,qp=12,prec=fp16 --cuda-schedule sync --output "E:\Temp\fp16_2021-01-09@10_23_11_7910_01.264"
```
The processing works:
```
--------------------------------------------------------------------------------
E:\Temp\1632_2021-01-09@10_23_11_7910_01.264
--------------------------------------------------------------------------------
NVEncC (x64) 5.24 (r1764) by rigaya, Dec 30 2020 12:27:24 (VC 1928/Win/avx2)
OS Version     Windows 10 x64 (19042)
CPU            AMD Ryzen 7 1800X Eight-Core Processor (8C/16T)
GPU            #0: GeForce GTX 1070 Ti (2432 cores, 1683 MHz)[PCIe3x16][461.09]
NVENC / CUDA   NVENC API 11.0, CUDA 11.2, schedule mode: sync
Input Buffers  CUDA, 32 frames
Input Info     avcuvid: H.265/HEVC, 640x480, 25/1 fps
AVSync         vfr
Vpp Filters    cspconv(p010 -> yv12)
               smooth: quality 3, qp 12, prec fp16
               cspconv(yv12 -> nv12)
Output Info    H.264/AVC high @ Level auto
               640x480p 1:1 25.000fps (25/1fps)
Encoder Preset default
Rate Control   VBR
Multipass      2pass-full
Bitrate        0 kbps (Max: 17500 kbps)
Target Quality auto
Initial QP     I:20  P:23  B:25
VBV buf size   auto
Lookahead      on, 16 frames, Adaptive I Insert
GOP length     250 frames
B frames       3 frames [ref mode: disabled]
Ref frames     3 frames
AQ             off
VUI            matrix:bt470bg
Others         mv:Q-pel cabac deblock adapt-transform:auto bdirect:auto

encoded 5000 frames, 2150.54 fps, 4.23 kbps, 0.10 MB
encode time 0:00:02, CPU: 6.8%, GPU: 60.3%, VE: 105.0%, VD: 44.3%, GPUClock: 1792MHz, VEClock: 1611MHz
frame type IDR   20
frame type I     20,  total size  0.00 MB
frame type P   1260,  total size  0.03 MB
frame type B   3720,  total size  0.07 MB
```
Problem is, the output is just green:
https://user-images.githubusercontent.com/843640/104088181-3340ee80-5265-11eb-9976-a63555d8c82a.mp4
Source I used: 
https://user-images.githubusercontent.com/843640/104088198-4e136300-5265-11eb-96e0-0fdb03ea1561.mp4

Using fp32 works fine.

Cu Selur

**Comment by Selur** [comment]
-> works fine with NVEncC (x64) 5.26 :)
Thanks

**Comment by rigaya** [comment]
Thanks for confirming the fix in fixed in [NVEnc 5.25](https://github.com/rigaya/NVEnc/releases/tag/5.25), it seems like that there were trouble with fp16 mode on Pascal Gen GPUs.
